posthog/models/subscription.py

```diff
@@ -0,0 +1,59 @@
+from django.contrib.postgres.fields import ArrayField
+from django.db import models
+
+
+class Subscription(models.Model):
+    """
+    Rather than re-invent the wheel, we are roughly following the iCalender format for recurring schedules
+    https://dateutil.readthedocs.io/en/stable/rrule.html
+
+    Also see https://jakubroztocil.github.io/rrule/ for a helpful visual demonstration
+    """
+
+    class SubscriptionTarget(models.TextChoices):
+        EMAIL = "email"
+        # SLACK = "slack"
+
+    class SubscriptionFrequency(models.TextChoices):
+        DAILY = "daily"
+        WEEKLY = "weekly"
+        MONTHLY = "monthly"
+        YEARLY = "yearly"
+
+    class SubscriptionByWeekDay(models.TextChoices):
+        MONDAY = "monday"
+        TUESDAY = "tuesday"
+        WEDNESDAY = "wednesday"
+        THURSDAY = "thursday"
+        FRIDAY = "friday"
+        SATURDAY = "saturday"
+        SUNDAY = "sunday"
+
+    # Relations - i.e. WHAT are we exporting?
+    team: models.ForeignKey = models.ForeignKey("Team", on_delete=models.CASCADE)
+    dashboard = models.ForeignKey("posthog.Dashboard", on_delete=models.CASCADE, null=True)
+    insight = models.ForeignKey("posthog.Insight", on_delete=models.CASCADE, null=True)
+
+    # Subscription type (email, slack etc.)
+    title: models.CharField = models.CharField(max_length=100, null=True, blank=True)
+    target_type: models.CharField = models.CharField(max_length=10, choices=SubscriptionTarget.choices)
+    target_value: models.CharField = models.CharField(max_length=65535)
```

Comment:
I guess "64k of emails" should be enough for anybody, right? 😅 

Should this be a textfield? Or is this a future problem?

Reply:
That makes sense. I'm just thinking if it should be a limitless `TextField`... or a 64k CharField

Reply:
Hmmm thats a good point. I think a TextField is more accurate tbh and less mysteriously specific than the value I gave